### ISSUE TITLE
Unification of sessions across machines (ssh)

### ISSUE THREAD

**Issue by TheButlah** [zellij-org/zellij#3237]
Hi! I've been using zellij for a while and love the project :)

The main limitation for me (other than windows support) is the inability to easily use zellij over ssh. I found #775 but it doesn't quite solve my issue.

I think what I would really like is for the ability to have zellij session manager track remote sessions. Then, I could take a particular tab in my current session, and set it to be a tab *in a remote session*. This would allow me to operate seamlessly over SSH without having to nest any zellij instances and getting the frames twice.

In other words, it would be great to (at least programatically) be able to have any tab be located either locally or remotely, and have a consistent way of referencing these things via the session manager.

Do you have any thoughts? Does this fit into zellij's intended architecture and use for sessions?

**Comment by giyany** [comment]
Hi! 
I hear you about the difficulty with ssh, this is also an experience that I don't like.
As in #775 , the idea is for a user to open a session in the remote server and in the future, allowing seamless integration between the local and remote. This is on the roadmap but not an easy problem to tackle. There's currently no ETA.

**Comment by heyakyra** [comment]
This would be great, and in addition to `ssh` it would be nice to handle connection persistence via existing projects like `et` and `mosh`: https://github.com/zellij-org/zellij/issues/5607

**Comment by giyany** [comment]
Since version 0.45.0, Zellij handles nested sessions so that running Zellij inside a Zellij pane (for example after ssh-ing into a remote machine and starting Zellij there) is a first-class workflow. Solved in #5417
